framework/include/constraints/EqualValueEmbededConstraint.h

```diff
@@ -0,0 +1,117 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef EqualValueEmbededConstraint_H
+#define EqualValueEmbededConstraint_H
+
+// MOOSE includes
+#include "NodeElemConstraint.h"
+// #include "ContactMaster.h"
+
+/**
+ * Contact models, currently only supports CM_GLUED
+ */
+enum ContactModel
+{
+  CM_INVALID,
+  CM_FRICTIONLESS,
+  CM_GLUED,
+  CM_COULOMB,
+};
+
+/**
+ * Contact formulations, currently only supports CF_KINEMATIC and CF_PENALTY
+ */
+enum ContactFormulation
+{
```

Comment:
Just keep kinematic and penalty.  Also rename to ConstraintFormulation or something like that.